compiler/packages/eslint-plugin-react-compiler/src/rules/ReactCompilerRule.ts

```diff
@@ -234,7 +234,7 @@ const rule: Rule.RuleModule = {
       nodeLoc: BabelSourceLocation,
       suppression: string,
     ): boolean {
-      const sourceCode = context.getSourceCode();
+      const sourceCode = context.sourceCode ?? context.getSourceCode();
       const comments = sourceCode.getAllComments();
```

Comment:
I think you can remove this declaration altogether since this function already closes over `sourceCode` from above.

```suggestion
      const comments = sourceCode.getAllComments();
```

Reply:
`sourceCode` in the larger scope is a string (`sourceCode.text`) rather than the full `sourceCode` object.  I could adjust that too, but this on its own wouldn't work.

Reply:
Yeah you can make the above variable the sourceCode object instead, and access `text` wherever else it was used previously

Reply:
Updated